ui/desktop/electron-app/src/cli/path.js

```diff
@@ -6,15 +6,20 @@
 const path = require('path');
 const { isWindows } = require('../helpers/platform.js');
 const isDev = require('electron-is-dev');
+const { existsSync } = require('node:fs');
 
 module.exports = {
-  // Returns boundary cli path
+  /**
+   * Returns Boundary CLI path if the CLI is built in or the Boundary binary name
+   * if not, so we assume boundary is available within user $PATH.
+   */
   path: () => {
-    const name = isWindows() ? 'boundary.exe' : 'boundary';
+    const binName = isWindows() ? 'boundary.exe' : 'boundary';
+    const builtInCliPath = isDev
+      ? path.resolve(__dirname, '..', '..', 'cli', binName)
+      : path.resolve(process.resourcesPath, 'cli', binName);
+    const isBuiltInCli = existsSync(builtInCliPath);
 
-    // The CLI will be located in a different location once packaged up in an ASAR
-    return isDev
-      ? path.resolve(__dirname, '..', '..', 'cli', name)
-      : path.resolve(process.resourcesPath, 'cli', name);
+    return isBuiltInCli ? builtInCliPath : binName;
```

Comment:
Perhaps a small change of `binName` to `binaryName` could help with readability. Also just to clarify: we return just the binary name when the cli folder does NOT exist because of the case where the user used the installer to download the binary ?

Reply:
@lisbet-alvarez 
Absolutley!! Change done! thanks for the suggestion.

In case the DC has a built in CLI we return the absolute path to it. Notice the path is different when you are in dev mode.

In case the DC does not have a built in CLI we return the binary name, assuming the boundary binary will be available at the users $Path, so boundary is present within users systems.

For more context: it is a "safe" assumption to made, since the DC with no CLI will be ONLY available for the installer. Will not be public as DC binary.

It is true, we will need to add a check, if the DC without CLI can't find it, it should gracefully let the user know. ATM it returns an error within the error modal, not ideal, but I will provide soon something better.